### PR TITLE
bootstrap carousel

### PR DESCRIPTION
**Pull request by zdubzdab**

### CODE REVIEW

File: app/assets/javascripts/welcome.coffee [hunk 1]
```diff
@@ -35,3 +35,17 @@ $ ->
       success: (data) ->
         $('#partial_news').html data
     event.preventDefault()
+
+  $('#myCarousel').carousel()
+  item = $('.carousel .item')
+  item.eq(0).addClass('active')
+  numberofSlides = $('.item').length
+  currentSlide = Math.floor(Math.random() * numberofSlides)
```
Comment (begin29) [inline app/assets/javascripts/welcome.coffee]: not sure about this one, why current slide is selected on random order?

File: app/assets/javascripts/welcome.coffee [hunk 2]
```diff
@@ -35,3 +35,17 @@ $ ->
       success: (data) ->
         $('#partial_news').html data
     event.preventDefault()
+
+  $('#myCarousel').carousel()
```
Comment (begin29) [inline app/assets/javascripts/welcome.coffee]: better move this few initialize rows to some method caled 'initializeCarusel' or some similar. and call it here.

File: app/assets/javascripts/welcome.coffee [hunk 3]
```diff
@@ -35,3 +35,17 @@ $ ->
...
+  item = $('.carousel .item')
+  item.eq(0).addClass('active')
+  numberofSlides = $('.item').length
+  currentSlide = Math.floor(Math.random() * numberofSlides)
+  $('.carousel-indicators li').each ->
```
Comment (begin29) [inline app/assets/javascripts/welcome.coffee]: this should be separate method called like `setActiveSlide` or some similar

File: app/controllers/welcome_controller.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 class WelcomeController < ApplicationController
 
   def index
-    @post = Tiding.where(main: true).last
+    @tiding = Tiding.where(main: true).last
```
Comment (begin29) [inline app/controllers/welcome_controller.rb]: add scope to `Tiding` model

File: app/views/welcome/_carousel.html.slim [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#myCarousel.carousel.slide data-ride="carousel"
+  ol.carousel-indicators
+    - @tiding.photos.each_with_index do |photo, index|
+      li.active data-slide-to="#{index}" data-target="#myCarousel"
+
+  .carousel-inner role="listbox"
+    - @tiding.photos.each do |photo|
+      .item
+        = link_to image_tag(photo.url(:thumb), alt: "Chania"), {class: "img-responsive", alt: "Responsive image"}
```
Comment (begin29) [inline app/views/welcome/_carousel.html.slim]: why `alt` here is static for each picture?

File: app/assets/javascripts/welcome.coffee (hunk 1 above)
Comment (zdubzdab) [inline app/assets/javascripts/welcome.coffee]: 'currentSlide' is just variable name I can rename it to "slide". Carusel starts from first photo  item.eq(0).addClass('active')

File: app/assets/javascripts/welcome.coffee (hunk 2 above)
Comment (zdubzdab) [inline app/assets/javascripts/welcome.coffee]: Should I create some folder "methods.coffee" and put this method into it?

File: app/assets/javascripts/welcome.coffee (hunk 3 above)
Comment (begin29) [inline app/assets/javascripts/welcome.coffee]: if it is used only here and not big, just create this method here, if it is repeated on few places than you need to separate it

File: app/assets/javascripts/my_methods.coffee [hunk 6]
```diff
@@ -0,0 +1,6 @@
+$.fn.initializeCarusel = ->
+  $(this).carousel()
+  item = $('.carousel .item')
+  item.eq(0).addClass('active')
+  numberofSlides = $('.item').length
+  currentSlide = Math.floor(Math.random() * numberofSlides)
```
Comment (begin29) [inline app/assets/javascripts/my_methods.coffee]: I steel see this `random logic`.

**Pull request merged**